src/app-layer-enip.c

```diff
@@ -343,7 +343,12 @@ static AppLayerResult ENIPParse(Flow *f, void *state, AppLayerParserState *pstat
         tx = ENIPTransactionAlloc(enip);
         if (tx == NULL)
             SCReturnStruct(APP_LAYER_OK);
-
+        if (flags & STREAM_TOSERVER)
+        {
+            tx->tx_data.detect_flags_tc |= APP_LAYER_TX_INSPECTED_FLAG;
```

Comment:
The parser is not supposed to set this field. Its reserved for the detection engine. We should probably make this struct opague. Can you submit a Suricata-Verify test for the issue you're seeing? I'm not sure I understand it.